recirq/otoc/loschmidt/tilted_square_lattice/analysis-walkthrough.ipynb

```diff
@@ -0,0 +1,373 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "agg_df3, gb_cols3 = analysis.groupby_all_except(\n",
+    "    df.drop('n_qubits', axis=1),\n",
+    "    y_cols=('instance_i', 'macrocycle_depth', 'q_area', 'success_probability'),\n",
+    "    agg_func=list,\n",
+    ")\n",
+    "agg_df3"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fit_df, exp_ansatz = analysis.fit_vs_macrocycle_depth(df)\n",
+    "fit_df"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Merging and Plotting\n",
+    "\n",
+    "To plot the mean+stddev data as well as visualizations of the fits, we merge (join) the two dataframes. Note that the groupby columns of the two dataframes are the same, so we can join on them:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print(gb_cols2)\n",
+    "print(gb_cols3)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "total_df = pd.merge(agg_df2, fit_df, on=gb_cols2)\n",
+    "total_df"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "colors = plt.get_cmap('tab10')\n",
+    "\n",
+    "for i, row in total_df.iterrows():\n",
+    "    plt.errorbar(\n",
+    "        x=row['macrocycle_depth'],\n",
+    "        y=row['success_probability_mean'],\n",
+    "        yerr=row['success_probability_std'],\n",
+    "        marker='o', capsize=5, ls='',\n",
+    "        color=colors(i),\n",
+    "        label=f'{row[\"width\"]}x{row[\"height\"]} ({row[\"n_qubits\"]}q) {row[\"processor_id\"]}; f={row[\"f\"]:.3f}'\n",
+    "    )\n",
+    "    \n",
+    "    xx = np.linspace(np.min(row['macrocycle_depth']), np.max(row['macrocycle_depth']))\n",
+    "    yy = exp_ansatz(xx, a=row['a'], f=row['f'])\n",
+    "    plt.plot(xx, yy, ls='--', color=colors(i))\n",
+    "    \n",
+    "plt.legend(loc='best')\n",
+    "plt.yscale('log')\n",
+    "plt.xlabel('Macrocycle Depth')\n",
+    "plt.ylabel('Success Probability')\n",
+    "plt.tight_layout()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Fit vs. \"Quantum Area\"\n",
+    "\n",
+    "In a local depolarizing model, we expect success to decay exponentially vs. circuit depth times the number of qubits. By defining this new quantity, we can perform a fit as before. The following cell shows the groupby operation used in `analysis.fit_vs_q_area`."
```

Comment:
I would make this sentence a bit more clearer and actually define quantum area.  Maybe:

By defining circuit depth times the number of qubits as "quantum area", we can perform a fit as before.

Reply:
how's this:

## Fit vs. "Quantum Area"

In a local depolarizing model, we expect success to decay exponentially in circuit depth and the number of qubits. We define a quantity called quantum area (`q_area`) which is the circuit width (i.e. number of qubits) multiplied by its depth. This is the number of operations in the circuit (also including any idle operations).

By defining this new quantity, we can fit a curve of fidelity vs. quantum area.. The following cell shows the groupby operation used in `analysis.fit_vs_q_area`.